### PR TITLE
NAS-124666 / 24.04 / Create new tickets with FeedbackDialog v2

### PR DESCRIPTION
**Pull request by denysbutenko**
Merge Feedback Dialog with FileTicketForms using multiple components.

For testing, try to rate a page or file a ticket using the topbar button or file ticket button on the support card.

### CODE REVIEW

**Review by RehanY147 — CHANGES_REQUESTED** [review]
We should
- Divide `FeedbackDialogComponent` into smaller components for actions and FormControls.
- In my opinion, we can use *ngIf for file ticket forms and just listen to form changes on those components instead of using complicated rendering logic. However, the current implementations works as well.
- On the "Report a Bug", we should hide the rest of the form if it's unusable until the Jira token is ready.
- Use a select instead of a button group to switch between forms. It looks like tab switching as it is and we are not too keen on that as a team preference. Would like @undsoft's opinion
- Furthermore, I think the option to report a problem or make a suggestion should be a radio-button or a select. Tabs are unnecessary since they have the exact same form. And having the option to switch between leaving a review or reporting a problem shouldn't be bundled together into one dialog. Let's have a discussion about the designs here.

**Comment by denysbutenko** [comment]
> We should
> 
> * Divide `FeedbackDialogComponent` into smaller components for actions and FormControls.
> * In my opinion, we can use *ngIf for file ticket forms and just listen to form changes on those components instead of using complicated rendering logic. However, the current implementations works as well.

Okay, I will consider that option and note it for further improvement.

> * On the "Report a Bug", we should hide the rest of the form if it's unusable until the Jira token is ready.

The only one field that depends on the Jira token is the Category select. Let's keep it disabled until the Jira token is ready.

> * Furthermore, I think the option to report a problem or make a suggestion should be a radio-button or a select. Tabs are unnecessary since they have the exact same form. We should offer a button that says something like "Facing a problem or have a suggestion for us? Create a ticket!". Or something to that effect. And if the user clicks that button or link etc, the dialog should completely switch to the ticket form. Or even redirect to a proper form page instead of just having a dialog be such a complicated function.

This sounds like what we currently have in the master branch. The idea is to merge three forms into one. [NAS-124484](https://ixsystems.atlassian.net/browse/NAS-124484) has more details.

**Comment by undsoft** [comment]
> The only one field that depends on the Jira token is the Category select. Let's keep it disabled until the Jira token is ready
Category won't be there in the final implementation. We want user not to have to select this.

File: src/app/modules/ix-feedback/file-ticket-form/file-ticket-form.component.spec.ts [hunk 1]
```diff
@@ -81,62 +72,40 @@ describe('FileTicketFormComponent', () => {
...
   beforeEach(() => {
     spectator = createComponent();
     loader = TestbedHarnessEnvironment.loader(spectator.fixture);
-    ws = spectator.inject(WebSocketService);
   });
 
-  it('loads ticket categories using api token when form is opened', async () => {
-    const form = await loader.getHarness(IxFormHarness);
-    const values = await form.getValues();
+  it.skip('loads ticket categories using api token when token is provided', async () => {
```
[the remark this answers is not in the document]
Comment (denysbutenko) [reply src/app/modules/ix-feedback/file-ticket-form/file-ticket-form.component.spec.ts]: For unknown reasons, `support.fetch_categories` is not called when a token is provided, and the Category select is still empty. Let's keep it as is. We've planned changes for the category shortly.

**Review by undsoft — APPROVED** [review]
There are still more things to do and submitting bugs doesn't really work, but I'd like to merge to enable more people to work on this.

**Pull request merged**